src/plugins/application_config/server/types.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { IScopedClusterClient } from 'src/core/server';
+
+export interface ApplicationConfigPluginSetup {
+  getConfigurationClient: (inputOpenSearchClient: IScopedClusterClient) => ConfigurationClient;
+  registerConfigurationClient: (inputConfigurationClient: ConfigurationClient) => void;
+}
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface ApplicationConfigPluginStart {}
+
+/**
+ * The interface defines the operations against the application configurations at both entity level and whole level.
+ *
+ */
+export interface ConfigurationClient {
```

Comment:
nit: In order to follow naming convention let's change it to `IConfigurationClient`.

Reply:
I did a quick search in the OSD code base. https://github.com/search?q=repo%3Aopensearch-project%2FOpenSearch-Dashboards+%22export+interface%22&type=code&p=1

Looks like most of the interfaces just have the regular name without "I" prefix. Let me know if there is a particular reason for this. Otherwise, I will leave the name as it is.